src/utils/cleanConstraint.ts

```diff
@@ -0,0 +1,29 @@
+import { singleValueOperators } from 'constants/operators';
+import { IConstraint } from 'interfaces/strategy';
+import { oneOf } from './one-of';
+
+const VALUES = 'values';
+const VALUE = 'value';
+
+export const cleanConstraint = (constraint: IConstraint): IConstraint => {
```

Comment:
Could mark the argument as Readonly to underline that this method won't modify the original:

```
export const cleanConstraint = (constraint: Readonly<IConstraint>): IConstraint => {
```

Reply:
Sounds good.